app/src/main/java/org/jellyfin/androidtv/details/DetailsOverviewPresenter.kt

```diff
@@ -119,34 +122,36 @@ class DetailsOverviewPresenter : RowPresenter() {
 			viewHolder.subtitle.visibility = View.GONE
 		}
 
-		// rating
-		if (item is Movie) { //todo move those properties to baseitem or something
+		if (item is Movie) {
 			if (item.productionYear != null) {
 				viewHolder.year.text = item.productionYear.toString()
 				viewHolder.year.visibility = View.VISIBLE
-			} else {
-				viewHolder.year.visibility = View.GONE
 			}
+		}
 
+		if (item is Episode) {
+			if (item.premiereDate != null) {
+				val format = DateFormat.getDateFormat(context)
+				viewHolder.year.text = format.format(item.premiereDate)
+				viewHolder.year.visibility = View.VISIBLE
+			}
+		}
+
+		// rating
+		if (item is Ratable) {
 			if (item.officialRating != null) {
```

Comment:
I can't post it as code suggestion because of the deleted lines but try to use the [not-null assertion operator](https://kotlinlang.org/docs/reference/null-safety.html#the--operator) as little as possible. Same applies for the other places you made the same change.
```kotlin
item.officialRating?.let { rating -> 
    viewHolder.officialRating.text = rating
    viewHolder.officialRating.visibility = View.VISIBLE
} ?: run {
    viewHolder.officialRating.visibility = View.GONE
}
````

Reply:
Will do